### PR TITLE
Config UI

### LINKED ISSUE #11: GUI-based configuration
It would be nice if all the configuration was done through a GUI window rather than through chat commands.

### PR DESCRIPTION
**Pull request by leatherface-dev**
Re-wrote the configuration code to use the UI instead of /slash commands.
I might have adjusted some of the project configs, so please feel free to revert if you need.
BTW, if desired, I could quickly add the DalamudPackager as well.. just leave a comment.

Closes #11

### CODE REVIEW

**Comment by tesu** [comment]
> BTW, if desired, I could quickly add the DalamudPackager as well.. just leave a comment.

Sure, go for it.

File: PennyPincher.cs [hunk 1]
```diff
@@ -74,173 +78,118 @@ public PennyPincher(DalamudPluginInterface pluginInterface, ChatGui chat, Comman
...
+            configuration.verbose = configVerbose;
+            
+            PluginInterface.SavePluginConfig(configuration);
         }
-
+        
         private bool Retainer()
         {
-            return (this.getFilePtr == null) ? false : Marshal.ReadInt64(this.getFilePtr(7), 0xB0) != 0;
+            return (getFilePtr != null) && Marshal.ReadInt64(getFilePtr(7), 0xB0) != 0;
```
Comment (tesu) [inline PennyPincher.cs:166]: Good catch. Can't believe I wrote the original nonsense.

File: PennyPincher.cs [hunk 2]
```diff
@@ -74,173 +78,118 @@ public PennyPincher(DalamudPluginInterface pluginInterface, ChatGui chat, Comman
...
+            ImGui.Begin($"{Name} Config", ref _config);
+            
+            ImGui.InputInt("Minimum Price", ref configMin);
+            ImGui.TextWrapped("Sets a minimum value to be copied. <min> cannot be below 1.");
+            
+            ImGui.InputInt("Mod", ref configMod);
+            ImGui.TextWrapped("Adjusts base price by subtracting <price> % <mod> from <price> before subtracting <delta>.\nThis makes the last digits of your posted prices consistent.");
+            
+            ImGui.InputInt("Delta", ref configDelta);
+            ImGui.TextWrapped("Sets the undercutting amount to be <delta>.");
```
Comment (tesu) [inline PennyPincher.cs:112]: Can you reorder this so that's delta, min, mod? I think delta is most important and should be at the top.

**Review by tesu — CHANGES_REQUESTED** [review]
Great stuff, thanks for working on this / cleaning it up.

File: PennyPincher.cs (hunk 2 above)
Comment (leatherface-dev) [reply PennyPincher.cs]: > 
> 
> Can you reorder this so that's delta, min, mod? I think delta is most important and should be at the top.
Done!

**Comment by leatherface-dev** [comment]
> 
> 
> > BTW, if desired, I could quickly add the DalamudPackager as well.. just leave a comment.
> 
> Sure, go for it.

Done!
I updated some of the project metadata (readme, json, version) accordingly as well, so please have a look.

**Review by tesu — APPROVED** [review]
Great, thanks!

**Pull request merged**